RecoEgamma/EgammaElectronProducers/plugins/TrackingRegionsFromSuperClustersProducer.h

```diff
@@ -239,14 +258,21 @@ getVtxPos(const edm::Event& iEvent,double& deltaZVertex)const
   //we fall back to beamspot mode
   auto beamSpotHandle = getHandle(iEvent,beamSpotToken_);
   const reco::BeamSpot::Point& bsPos = beamSpotHandle->position();
-  //SH: this is what SeedFilter did, no idea what its trying to achieve....
-  const double sigmaZ = beamSpotHandle->sigmaZ();
-  const double sigmaZ0Error = beamSpotHandle->sigmaZ0Error();
-  deltaZVertex = 3*std::sqrt(sigmaZ*sigmaZ+sigmaZ0Error*sigmaZ0Error);
-  //  std::cout <<"z "<<bsPos.z()<<" deltaZ "<<deltaZVertex<<std::endl;
-  //deltaZVertex = 30.;
-  return GlobalPoint(bsPos.x(),bsPos.y(),bsPos.z());
-  //return GlobalPoint(bsPos.x(),bsPos.y(),0);
+  
+  if(useZInBeamspot_){
```

Comment:
The comment at line 257-258 is not true any more: now if the vertex collection is empty we fall back to "default" (not beamspot) mode. 

Depending on what is your original intention, either modify that comment, or  put an "else" with the same content as of lines 271-272 after L254